base/socket.jl

```diff
@@ -575,14 +575,14 @@ function uv_getaddrinfocb(req::Ptr{Void}, status::Cint, addrinfo::Ptr{Void})
     nothing
 end
 
-function getaddrinfo(cb::Function, host::ASCIIString)
+function getaddrinfo(cb::Function, host::String)
     callback_dict[cb] = cb
     uv_error("getaddrinfo",ccall(:jl_getaddrinfo, Int32, (Ptr{Void}, Cstring, Ptr{UInt8}, Any, Ptr{Void}),
                                  eventloop(), host, C_NULL, cb, uv_jl_getaddrinfocb::Ptr{Void}))
 end
 getaddrinfo(cb::Function, host::AbstractString) = getaddrinfo(cb,ascii(host))
 
-function getaddrinfo(host::ASCIIString)
+function getaddrinfo(host::String)
```

Comment:
There's a small inconsistency here and above: the method for `AbstractString` just below calls `ascii`. Should check whether `uv_getaddrinfo` is safe with non-ASCII input, and if not validate the string.